pkg/api/controllers/metrics.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2019 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound service.
+
+*/
+
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/golang/glog"
+	"github.com/opensds/opensds/pkg/api/policy"
+	c "github.com/opensds/opensds/pkg/context"
+	"github.com/opensds/opensds/pkg/controller/client"
+	"github.com/opensds/opensds/pkg/model"
+	pb "github.com/opensds/opensds/pkg/model/proto"
+	. "github.com/opensds/opensds/pkg/utils/config"
+	"golang.org/x/net/context"
+)
+
+func NewMetricsPortal() *MetricsPortal {
+	return &MetricsPortal{
+		CtrClient: client.NewClient(),
+	}
+}
+
+type MetricsPortal struct {
+	BasePortal
+
+	CtrClient client.Client
+}
+
+func (m *MetricsPortal) GetMetrics() {
+	if !policy.Authorize(m.Ctx, "metrics:get") {
+		return
+	}
+	ctx := c.GetContext(m.Ctx)
+	var getMetricSpec = model.GetMetricSpec{
+		BaseModel: &model.BaseModel{},
+	}
+
+	// Unmarshal the request body
+	if err := json.NewDecoder(m.Ctx.Request.Body).Decode(&getMetricSpec); err != nil {
+		errMsg := fmt.Sprintf("parse get metric request body failed: %s", err.Error())
+		m.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}
+
+	m.SuccessHandle(StatusOK, nil)
+
+	// NOTE:The real volume creation process.
```

Comment:
These code is designed for async operation like create volume, so there is no need to apply it to get metrics method.

Reply:
have removed this 
```m.SuccessHandle(StatusOK, nil)```
but retained this, to return the response
```m.SuccessHandle(StatusOK, []byte(res.GetResult().GetMessage()))```